cmd/common.go

```diff
@@ -55,7 +55,15 @@ func AddCommonFlags(cmd *cobra.Command, cfg *CommonConfig) {
 	cmd.PersistentFlags().StringVar(&cfg.StoreEndpoints, "store-endpoints", "", "a comma-delimited list of store endpoints (use https scheme for tls communication) (defaults: http://127.0.0.1:2379 for etcd, http://127.0.0.1:8500 for consul)")
 	cmd.PersistentFlags().StringVar(&cfg.StorePrefix, "store-prefix", common.StorePrefix, "the store base prefix")
 	cmd.PersistentFlags().StringVar(&cfg.StoreCertFile, "store-cert-file", "", "certificate file for client identification to the store")
+
 	cmd.PersistentFlags().StringVar(&cfg.StoreKeyFile, "store-key", "", "private key file for client identification to the store")
+	cmd.PersistentFlags().MarkHidden("store-key")
+	cmd.PersistentFlags().MarkDeprecated("store-key", "use --store-key-file instead")
+
+	if !cmd.PersistentFlags().Changed("store-key") {
```

Comment:
Calling Changed() in the init method (since AddCommonFlags is called inside init() ) will always return false. My idea was to add both flags here (with store-key marked as deprecated) and in `CheckCommonConfig` verify (using the Changed() method) that not both options are changed or return an error.

Reply:
@sgotti ok, I will try to do it today.